tools/adventure-pack/src/iteratorPrototypeFilter.ts

```diff
@@ -0,0 +1,26 @@
+import { iteratorPrototype } from "./iteratorPrototype";
+import { iteratorToIterable } from "./iteratorToIterable";
+
+declare global {
+  interface Iterator<T> {
+    filter(
+      callbackfn: (value: T, index: number, array: T[]) => any,
+      thisArg?: any,
+    ): Generator<T, void, undefined>;
+  }
+}
+
+iteratorPrototype.filter ??= function* <T>(
+  this: Iterator<T>,
+  callbackFn: (element: T, index: number, array: T[]) => any,
+) {
+  let index = 0;
+  const iterable = iteratorToIterable(this);
+  const array = Array.from(iterable);
```

Comment:
Thanks for thinking about the third argument, but converting the iterable to an array like this somewhat undoes the advantage of the iterator. The reason to prefer an iterator/generator is that it's lazy and uses less memory by not creating an intermediate array like all the array methods.

I think the upcoming [Iterator.prototype.filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator/filter) doesn't pass any third argument, so it's ok if we don't either.